.github/workflows/docker-sandbox.yml

```diff
@@ -73,22 +73,24 @@ jobs:
         password: "${{ secrets.DockerPassword }}"
         image_tag: latest,${{ steps.get_version.outputs.VERSION }}
         context: ./docker
-        pull_image_and_stages: false
 
-    # - name: Clean up Docker environment
-    #   if: failure()
-    #   run: docker system prune --force
+    # If the above step fails, clean up the environment and build without a cache
+    - name: Clean up Docker environment
+      if: failure()
+      run: |
+        docker system prune --force --volumes
+        docker rmi $(docker images --format '{{.Repository}}:{{.Tag}}' | grep 'geoscienceaustralia')
 
-    # - name: Build and Push latest Docker image without a cache
-    #   if: failure()
-    #   uses: whoan/docker-build-with-cache-action@v4
-    #   with:
-    #     image_name: ${{ env.IMAGE_NAME }}
-    #     username: ${{ env.DOCKER_USER }}
-    #     password: "${{ secrets.DockerPassword }}"
-    #     image_tag: latest,${{ steps.get_version.outputs.VERSION }}
-    #     pull_image_and_stages: false
-    #     context: ./docker
+    - name: Build and Push latest Docker image without a cache
+      if: failure()
+      uses: whoan/docker-build-with-cache-action@v4
+      with:
+        image_name: ${{ env.IMAGE_NAME }}
+        username: ${{ env.DOCKER_USER }}
+        password: "${{ secrets.DockerPassword }}"
+        image_tag: latest,${{ steps.get_version.outputs.VERSION }}
+        pull_image_and_stages: false
+        context: ./docker
```

Comment:
So in this mode does it still push stages that it builds to dockerhub? So that next time cache can be used.

Reply:
Yeah, there's two flags. One for pulling and one for pushing.

This step runs on the `master` branch only and will push the stages and the actual image.

Reply:
The test build sets the flag `push_image_and_stages: false` so that it doesn't push for PRs and pushes.